src/main/java/org/generationcp/middleware/service/impl/study/StudyGermplasmServiceImpl.java

```diff
@@ -66,18 +76,45 @@ public StudyGermplasmServiceImpl(final HibernateSessionProvider sessionProvider)
 		return studyGermplasmDtos;
 	}
 
+	@Override
+	public List<StudyEntryDto> getStudyEntries(final int studyId, final StudyEntrySearchDto.Filter filter, final Pageable pageable) {
+
+		final Integer plotDatasetId =
+			datasetService.getDatasets(studyId, new HashSet<>(Arrays.asList(DatasetTypeEnum.PLOT_DATA.getId()))).get(0).getDatasetId();
+
+		final List<MeasurementVariable> entryDescriptors =
+			this.datasetService.getObservationSetVariables(plotDatasetId, Lists
+				.newArrayList(VariableType.GERMPLASM_DESCRIPTOR.getId()));
+
+		//Remove the ones that are stored in stock and that in the future will not be descriptors
+		final List<Integer> termsToRemove = Lists
+			.newArrayList(TermId.ENTRY_CODE.getId(), TermId.DESIG.getId(), TermId.ENTRY_NO.getId(), TermId.GID.getId(),
+				TermId.OBS_UNIT_ID.getId(), TermId.STOCKID.getId());
+		for (Iterator<MeasurementVariable> i = entryDescriptors.iterator(); i.hasNext(); ) {
```

Comment:
This for loop can be replaced by `Collections.removeIf`:
`entryDescriptors.removeIf(entry -> termsToRemove.contains(entry.getTermId()));`